pkg/codegen/go/gen_program.go

```diff
@@ -102,6 +95,58 @@ func GenerateProgram(program *pcl.Program) (map[string][]byte, hcl.Diagnostics,
 	return files, g.diagnostics, nil
 }
 
+func GenerateProgram(program *pcl.Program) (map[string][]byte, hcl.Diagnostics, error) {
+	packages, contexts := map[string]*schema.Package{}, map[string]map[string]*pkgContext{}
+	for _, pkg := range program.Packages() {
+		packages[pkg.Name], contexts[pkg.Name] = pkg, getPackages("tool", pkg)
+	}
+
+	g := &generator{
+		program:             program,
+		packages:            packages,
+		contexts:            contexts,
+		spills:              &spills{counts: map[string]int{}},
+		jsonTempSpiller:     &jsonSpiller{},
+		ternaryTempSpiller:  &tempSpiller{},
+		readDirTempSpiller:  &readDirSpiller{},
+		splatSpiller:        &splatSpiller{},
+		optionalSpiller:     &optionalSpiller{},
+		scopeTraversalRoots: codegen.NewStringSet(),
+		arrayHelpers:        make(map[string]*promptToInputArrayHelper),
+	}
+
+	return generateProgram(program, g)
+}
+
+func GenerateProgramWithOpts(program *pcl.Program, opts ...GenerateProgramOption) (
```

Comment:
Maybe spell out `Opts` as `Options` to match what we did elsewhere like the following?

https://github.com/pulumi/pulumi/blob/6a8f7be1d548d7b3c02fadecc238933722fb0110/pkg/resource/provider/component_provider.go#L51